Sources/WhoopDIKit/Service/ServiceDictionary.swift

```diff
@@ -26,7 +26,11 @@ public final class ServiceDictionary<Value> {
             valuesByType[key] = newValue
         }
     }
-    
+
+    public func copy() -> Self {
+        return Self(valuesByType: self.valuesByType)
```

Comment:
I think we would want to create a new instance of the dictionary here which takes the values in, otherwise we have the same reference to a dict under the hood which could have issues with races I think (since this is a class not a struct)

Reply:
Since we are passing the dictionary into a _new_ class, we should have a copy. Technically copy on write will make it the same instance, but it should be fine since those updates are atomic. What would be dangerous would be passing the ServiceDictionary itself, since that has a pointer to the actual same dictionary in multiple places and is definitely not sendable